docs/learn/migrate/solana.mdx

```diff
@@ -1,8 +1,689 @@
+
+# Add Soroban dependencies to Cargo.toml
+```
+
+## Migrating Your First Contract
+
+Let's walk through the migration process step by step. We'll start with a simple example and gradually build up to more complex patterns.
+
+### Step 1: Update Dependencies
+
+First, you'll need to replace your Solana dependencies with Stellar equivalents. Here's what your `Cargo.toml` should look like:
+
+```toml
+[package]
+name = "my-soroban-contract"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+soroban-sdk = "$VERSION"
+
+[dev_dependencies]
+soroban-sdk = { version = "$VERSION", features = ["testutils"] }
+```
+
+:::note **Important**:
+
+Replace `$VERSION` with the actual latest version from [crates.io](https://crates.io/crates/soroban-sdk). As of the latest Stellar documentation, the current stable version is v20.4.0, but you should always check for the most recent release.
+
+:::
+
+The release profile configuration is crucial for keeping your contract size under the 64KB limit:
+
+```toml
+[profile.release]
+opt-level = "z"
+overflow-checks = true
+debug = 0
+strip = "symbols"
+debug-assertions = false
+panic = "abort"
+codegen-units = 1
+lto = true
+
+[profile.release.build-override]
+opt-level = 3
+incremental = false
+codegen-units = 1
+```
+
+### Step 2: Convert Program Entry Point
+
+The biggest change you'll notice is how functions are defined and called.
+
+**Before (Solana):**
+
+```rust
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint,
+    entrypoint::ProgramResult,
+    msg,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+entrypoint!(process);
+
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let payer = next_account_info(accounts_iter)?;
+
+    msg!("Hello from Solana!");
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[contract]
+pub struct HelloContract;
+
+#[contractimpl]
+impl HelloContract {
+    pub fn hello(env: Env, to: Symbol) -> Vec<Symbol> {
+        vec![&env, symbol_short!("Hello"), to]
+    }
+}
+```
+
+The Stellar version is much more intuitive - just define your struct, implement the functions you want, and the framework handles the rest.
+
+### Step 3: Convert Account Handling
+
+Solana requires you to manually manage accounts and validate ownership. Stellar simplifies this significantly.
+
+**Before (Solana):**
+
+```rust
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let user_account = next_account_info(accounts_iter)?;
+    let data_account = next_account_info(accounts_iter)?;
+
+    // Validate account ownership
+    if user_account.owner != program_id {
```

Comment:
Don't you need to verify that user_account `is_signer`? Or, maybe, use anchor as suggested in the documentation (e.g. https://solana.com/developers/courses/program-security/signer-auth)? I'm very vaguely familiar with Solana, but it does seem to me like this example is unsafe. Please correct me if I'm wrong.